### ISSUE TITLE
How to use `LiveReloadPlugin` with Angular 8?

### ISSUE THREAD

**Issue by seanmarthur** [just-jeb/angular-builders#335]
**Describe the bug**
I've recently upgraded to Angular 8 and I have a bit of a special case. I need to be able to output a custom index file to a mapped drive and use the webpack `LiveReloadPlugin` for development.

I've seen the [migration guide](https://github.com/meltedspark/angular-builders/blob/master/MIGRATION.MD) and mention about the index file, but I'm not quite sure how to proceed.

I have this `extra-webpack.config.js` file:

```
module.exports = {
  watch: true,
  output: {
    path: mappedDrive // like "Z:\"
  },
  plugins: [
    new LiveReloadPlugin({
      protocol: 'http',
      hostname: 'localhost',
      appendScriptTag: true
    })
  ]
};
```

I've tried setting the relevant `outputPath` property in angular.json to match the mapped drive, and everything gets output to the proper location *except* for my index file. Any ideas?

angular.json
```
      "architect": {
        "custom-dev": {
          "builder": "@angular-builders/custom-webpack:browser",
          "options": {
            "customWebpackConfig": {
              "path": "./extra-webpack.config.js"
            },
            "deleteOutputPath": false,
            "outputPath": "Z:\\",
```

**Builder:**
`"@angular-builders/custom-webpack": "8.0.2",`

**Libraries**
`"@angular-devkit/build-angular": "~0.800.0",`

**Additional context**
Basically, I need to be able to run a dev build which is not running a local webpack-dev-server. I want to inject and use the `LiveReloadPlugin` so I can get a similar experience with auto-reload when the bundles update. I'm open to other options, this has been working for me since the release of this custom-webpack builder.

**Comment by meltedspark** [comment]
Unfortunately you can't modify the generation of `index.html` with webpack since it is not generated as part of webpack build anymore (just like stated in migration guide).

**Comment by seanmarthur** [comment]
Thanks for your response.

We build our applications and deploy them to a COTS CMS system - we aren't able to fully emulate a local dev environment to use the dev-server in all instances (that part is also broken for Angular 8, but that's another story...) Some types of apps need to be developed within the full context of the CMS system, so we deploy to a mapped location directly to the server, and load the deployed app in the browser. With the livereload plugin, we could get a decent dev experience where each build would deploy the full app to the server and auto-reload.

I guess if I had some way to run code as the last step of a watcher build (around the 'emit' step) to perform the index file update/move, that would work, I could just manually inject the livereload script and copy the file to the mapped drive...

**Comment by meltedspark** [comment]
The new way of modifying the `index.html` is implementing a transform function. It is not released yet (will be released soon). After it is released I'll have to support this functionality in the builder.  
As far as I understand you don't need to customize the `index.html` but you do need to signal the live reload server that it has to reload the chunk once it is changed.
You can implement a transform that basically does nothing but tells the server to reload (should probably be delayed by a few hundreds of milliseconds). Anyways you won't be able to do that until Angular release `@next` version and I support it in the builder.  
Did I get everything right?

**Comment by seanmarthur** [comment]
There are two problems as I see it:

1. If `outputPath` is an absolute path (like Z:\\) - the index file does not end up in that location. All other app files do.
2. I need to manipulate the content of the index file to inject the livereload script - I could probably technically do this in the Angular app itself (if devMode -> append script tag to head) - I'd rather have it be part of the build chain tho.

I'll hold off for a few weeks to see if things improve. Thanks for your work on this project - the ability to muck with the webpack build while keeping the goodness of the CLI has been an insane boost to our team.

**Comment by meltedspark** [comment]
1. That would probably be a bug in Angular CLI, try to file an issue in their repo.
2. Once they release the `@next` version and I support it in the builder you'll be able to append the script tag via the transform function.

**Comment by meltedspark** [comment]
[`indexTransform`](https://github.com/meltedspark/angular-builders/tree/master/packages/custom-webpack#index-transform) is available in @next version. Please let me know if it helps you and if I can close this issue.

**Comment by seanmarthur** [comment]
Thank you - I finally was able to revisit this. After re-executing the upgrade, my original issue seems to be gone - my custom index template is being used, and the webpack plugins that modify it seem to work just fine. The new 'bundle splitting' thing is what seems to be missing at this point. 

I am getting two different index files generated - one processed straight from the CLI (and output as the only file directly to dist), and the other which appears to be from the webpack build (output to the path specified in the custom webpack config - along with the rest of the app). This isn't actually a problem as it only occurs during development. The full output from ng build all appears correct.

All seems to be good, although this feels like a bug is going on somewhere. There are definitely two 'paths' that index file processing is going down...

We can close this for now - thank you again!
